dev/lint-python

```diff
@@ -96,6 +98,32 @@ fi
 
 rm "$PEP8_REPORT_PATH"
 
+# Check that the documentation builds acceptably, skip check if sphinx is not installed.
+if hash "$SPHINXBUILD" 2> /dev/null; then
+  cd python/docs
+  make clean
+  export SPHINXOPTS=""
+  make linkcheck &>> "$SPHINX_REPORT_PATH"  || lint_status=1
+  make clean
+  # Treat warnings as errors so we stop correctly
+  export SPHINXOPTS="-a -W"
```

Comment:
Not necessary to `export`. We can change the next line to

~~~
SPHNIXOPTS="-a -W" make html ...
~~~

L116 is similar.